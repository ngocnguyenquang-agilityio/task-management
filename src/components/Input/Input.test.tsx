import { render, screen, fireEvent } from '@testing-library/react';

// Components
import { Input } from './';

describe('Input component', () => {
  const handleChange = jest.fn();

  test('renders the default Input', () => {
    expect(render(<Input onChange={handleChange} />)).toMatchSnapshot();
  });

  test('Test the correct variant class', () => {
    render(<Input onChange={handleChange} variant="outline" />);
    expect(screen.getByRole('textbox')).toHaveClass(
      'bg-transparent text-zinc-500',
    );
  });

  test('Test customClass', () => {
    render(<Input customClass="custom-class" onChange={handleChange} />);
    expect(screen.getByRole('textbox')).toHaveClass('custom-class');
  });

  test('Test handles change events', () => {
    render(<Input onChange={handleChange} />);
    fireEvent.change(screen.getByRole('textbox'), {
      target: { value: 'Test input' },
    });
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  test('Test when the disabled prop is set', () => {
    render(<Input disabled onChange={handleChange} />);
    expect(screen.getByRole('textbox')).toBeDisabled();
  });
});
