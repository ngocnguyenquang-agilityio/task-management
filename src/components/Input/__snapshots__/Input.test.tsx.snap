// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders the default Input 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="p-4 border border-zinc-300 focus:outline focus:outline-zinc-300 rounded-lg w-full dark:border-gray-700 dark:focus:outline-gray-700 dark:focus:border-gray-700 disabled:cursor-not-allowed disabled:opacity-50 bg-zinc-50 text-zinc-700 dark:bg-zinc-800 dark:text-white"
        placeholder="Type here"
        type="text"
        value=""
      />
    </div>
  </body>,
  "container": <div>
    <input
      class="p-4 border border-zinc-300 focus:outline focus:outline-zinc-300 rounded-lg w-full dark:border-gray-700 dark:focus:outline-gray-700 dark:focus:border-gray-700 disabled:cursor-not-allowed disabled:opacity-50 bg-zinc-50 text-zinc-700 dark:bg-zinc-800 dark:text-white"
      placeholder="Type here"
      type="text"
      value=""
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
